### PR TITLE
 [#187364741] A User should be able to view a specific item

### PR DESCRIPTION
**Pull request by Bananayosostene**
**What does this PR do?**
This PR enhances the application's functionality to allow users, both Sellers and Buyers, to view specific items. It ensures that the retrieval process is robust and secure, providing the appropriate response based on the user's role and the item's availability.

**Description of Task to be completed?**
The task involves implementing endpoints to retrieve the details of a specific item. Different scenarios are considered, including cases where the item is in a seller's collection, does not exist in the seller's collection, or when the requesting user is a buyer. Validation checks are implemented to ensure the integrity and security of the retrieval process.

**How should this be manually tested?**
Send a GET request to retrieve the details of a specific item.
Test various scenarios:
Verify the response when the target item exists in the seller's collection.
Check the response when the item does not exist in the seller's collection.
Test the response for buyer users, ensuring they can view available items.
Validate the response for scenarios where the request body fails validation checks or the token is not of the seller role.
Ensure appropriate error responses are returned when necessary.
Confirm that the response objects contain the correct status codes, messages, and item details.
Any background context you want to provide?
Implementing this feature improves the user experience by providing seamless access to item details based on user roles and item availability. It ensures that users, whether Sellers or Buyers, can efficiently retrieve the information they need.

**What are the relevant pivotal tracker/Trello stories?**
[#187364741]

**VIDEO**
https://www.loom.com/share/aab9d527f9f148ff9c0f3448b23c2fd2

**Questions:**
[If there are any questions or concerns, please feel free to raise them here.]

### CODE REVIEW

File: src/database/models/ItemModel.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      primaryKey: true,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    description: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
```
Comment (AngeloChristian1) [inline src/database/models/ItemModel.ts]: You can add other attributes like price, category, Images, expireDate,... and others

**Review by AngeloChristian1 — COMMENTED** [review]
Nice work, you can try to update the model by adding more columns for product

**Comment by Bananayosostene** [comment]
Please give me comment to this PR fill free also I did self review, Thank you!!!

**Review by Yvan-David — CHANGES_REQUESTED** [review]
I think instead of creating a getProduct controller as a file you should use product Controller file and creat there your functions 

also remember to use Product services file to access some functions instead of recreating them


Please remember to put some screenshots or video to see your swagger documentation

File: src/controllers/productController.ts [hunk 2]
```diff
@@ -111,6 +111,61 @@ export default class productController {
...
+        })
+      }
+
+      if (userRole === 'buyer' || userRole === 'admin') {
+        if (product.quantity > 0) {
+          return res.status(200).json({
+            status: 200,
+            message: 'Item retrieved successfully',
+            item: product,
+          })
```
Comment (EricNiyo73) [inline src/controllers/productController.ts]: The product which is unavailable ,bought  or expired are not allowed to be displayed to the buyer
And also I think the admin can view all the products posted for checking if the user has not posted the illegal products.

File: src/controllers/productController.ts [hunk 3]
```diff
@@ -111,6 +111,61 @@ export default class productController {
...
+    const productId = req.params.id
+
+    try {
+      const product = await Product.findByPk(productId)
+
+      if (!product) {
+        return res.status(404).json({ status: 404, error: 'Item not found' })
+      }
+
+      if (userRole === 'seller') {
```
Comment (EricNiyo73) [inline src/controllers/productController.ts]: Import **enum userRole from userModel** and use it instead

**Review by EricNiyo73 — CHANGES_REQUESTED** [review]

**Review by Yvan-David — APPROVED** [review]
Great Job
As shown in the video provided

**Review by EricNiyo73 — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**